includes/admin/home/notices/class-sensei-home-notices.php

```diff
@@ -54,6 +54,119 @@ public function __construct( Sensei_Home_Remote_Data_API $remote_data_api, strin
 	public function init() {
 		add_filter( 'sensei_show_admin_notices_' . $this->screen_id, '__return_false' );
 		add_filter( 'sensei_admin_notices', [ $this, 'add_update_notices' ], 10, 2 );
+		add_filter( 'sensei_admin_notices', [ $this, 'add_review_notice' ], 10, 2 );
+	}
+
+	/**
+	 * Return the answer the user given to the review.
+	 *
+	 * @param string $nonce The nonce to consider while checking the nonce value.
+	 * @return string An empty string, or '0', or '1'.
+	 */
+	private function get_review_answer( $nonce ) {
```

Comment:
Handing in the nonce here seems a bit odd to me...is there a strong reason to do so?

To be fair, I don't think I have a strong reason _not_ to do so, except that it doesn't seem necessary (i.e. the nonce string could be given by a constant instead). In fact, given that this is a GET request and the GET parameter isn't being used to alter any data, probably we don't need a nonce here at all 🤷‍♂️

Reply:
In this case, I don't see any serious issues with removing the nonce (the ones you outlined make sense, but I don't think they are likely to happen in practice or have high impact). But I also agree that it doesn't hurt to keep it.

If we keep it, I would personally still lean toward using a constant for the nonce action as I think it simplifies things (and is more conventional than needing to pass it around). The nonce for dismissing a notice ([in this class](https://github.com/Automattic/sensei/blob/trunk/includes/admin/class-sensei-admin-notices.php)) is a good example. Note that it doesn't need to be a dynamically created string (i.e. it doesn't need to be `$notice_id`), and probably makes more sense not to be.

Reply:
Given your suggestion to change things in frontend instead, I decided to abandon all things related to nonce here, and so I removed this method in [5ac5933085b5089c4f382f1168c18088860b4975](https://github.com/Automattic/sensei/commit/5ac5933085b5089c4f382f1168c18088860b4975). :)